sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_base.py

```diff
@@ -0,0 +1,289 @@
+    USER_AGENT_SYMBOL: Union[uamqp_types.AMQPSymbol, str, bytes]
+    PROP_PARTITION_KEY_AMQP_SYMBOL: Union[uamqp_types.AMQPSymbol, str, bytes]
+
+    @staticmethod
+    @abstractmethod
+    def build_message(**kwargs):
+        """
+        Creates a uamqp.Message or pyamqp.Message with given arguments.
+        :rtype: uamqp.Message or pyamqp.Message
+        """
+
+    @staticmethod
+    @abstractmethod
+    def build_batch_message(**kwargs):
+        """
+        Creates a uamqp.BatchMessage or pyamqp.BatchMessage with given arguments.
+        :rtype: uamqp.BatchMessage or pyamqp.BatchMessage
+        """
+
+    @staticmethod
+    @abstractmethod
+    def to_outgoing_amqp_message(annotated_message):
+        """
+        Converts an AmqpAnnotatedMessage into an Amqp Message.
+        :param AmqpAnnotatedMessage annotated_message: AmqpAnnotatedMessage to convert.
+        :rtype: uamqp.Message or pyamqp.Message
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_message_encoded_size(message):
+        """
+        Gets the message encoded size given an underlying Message.
+        :param uamqp.Message or pyamqp.Message message: Message to get encoded size of.
+        :rtype: int
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_remote_max_message_size(handler):
+        """
+        Returns max peer message size.
+        :param AMQPClient handler: Client to get remote max message size on link from.
+        :rtype: int
+        """
+
+    @staticmethod
+    @abstractmethod
+    def create_retry_policy(config):
+        """
+        Creates the error retry policy.
+        :param ~azure.eventhub._configuration.Configuration config: Configuration.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def create_link_properties(link_properties):
+        """
+        Creates and returns the link properties.
+        :param dict[bytes, int] link_properties: The dict of symbols and corresponding values.
+        :rtype: dict
+        """
+
+    @staticmethod
+    @abstractmethod
+    def create_connection(**kwargs):
+        """
+        Creates and returns the uamqp Connection object.
+        :keyword str host: The hostname, used by uamqp.
+        :keyword JWTTokenAuth auth: The auth, used by uamqp.
+        :keyword str endpoint: The endpoint, used by pyamqp.
+        :keyword str container_id: Required.
+        :keyword int max_frame_size: Required.
+        :keyword int channel_max: Required.
+        :keyword int idle_timeout: Required.
+        :keyword Dict properties: Required.
+        :keyword int remote_idle_timeout_empty_frame_send_ratio: Required.
+        :keyword error_policy: Required.
+        :keyword bool debug: Required.
+        :keyword str encoding: Required.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def close_connection(connection):
+        """
+        Closes existing connection.
+        :param connection: uamqp or pyamqp Connection.
```

Comment:
Does this need a type? and below too

Reply:
I didn't, just for the sake of time and b/c it's internal. But added an issue to the backlog to track: #26070.

Lmk if you think we should def fix these/get them in before the GA/release?

Reply:
Eventually yes, but doesn't need to be for this PR